server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentAction.java

```diff
@@ -266,20 +265,126 @@ private static void addBestValuesToMeasures(List<MeasureDto> measures, Component
     }
   }
 
-  private static ComponentWsRequest toComponentWsRequest(Request request) {
-    ComponentWsRequest componentWsRequest = new ComponentWsRequest()
+  private static ComponentRequest toComponentWsRequest(Request request) {
+    ComponentRequest componentRequest = new ComponentRequest()
       .setComponentId(request.param(DEPRECATED_PARAM_COMPONENT_ID))
       .setComponent(request.param(PARAM_COMPONENT))
       .setBranch(request.param(PARAM_BRANCH))
       .setAdditionalFields(request.paramAsStrings(PARAM_ADDITIONAL_FIELDS))
       .setMetricKeys(request.mandatoryParamAsStrings(PARAM_METRIC_KEYS))
       .setDeveloperId(request.param(PARAM_DEVELOPER_ID))
       .setDeveloperKey(request.param(PARAM_DEVELOPER_KEY));
-    checkRequest(!componentWsRequest.getMetricKeys().isEmpty(), "At least one metric key must be provided");
-    return componentWsRequest;
+    checkRequest(!componentRequest.getMetricKeys().isEmpty(), "At least one metric key must be provided");
+    return componentRequest;
   }
 
   private void checkPermissions(ComponentDto baseComponent) {
     userSession.checkComponentPermission(UserRole.USER, baseComponent);
   }
+
+  private static class ComponentRequest {
+    private String componentId;
+    private String componentKey;
+    private String component;
+    private String branch;
+    private List<String> metricKeys;
+    private List<String> additionalFields;
+    private String developerId;
+    private String developerKey;
+
+    /**
+     * @deprecated since 6.6, please use {@link #getComponent()} instead
+     */
+    @Deprecated
+    @CheckForNull
+    public String getComponentId() {
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)